e2e/libs/longhorn_deploy/longhorn_deploy.py

```diff
@@ -47,19 +48,20 @@ def install(self, install_stable_version):
         installed = self.longhorn.install(install_stable_version)
         if not installed:
             logging(f"Installing Longhorn failed")
-            time.sleep(self.retry_count)
+            time.sleep(self.retry_interval)
             assert False, "Installing Longhorn failed"
         self.longhorn.setup_longhorn_ui_nodeport()
         logging(f"Installed Longhorn")
 
-    def upgrade(self, upgrade_to_transient_version):
+    def upgrade(self, upgrade_to_transient_version, timeout):
         logging(f"Upgrading Longhorn to {'transient' if upgrade_to_transient_version else 'the latest'} version")
-        upgraded = self.longhorn.upgrade(upgrade_to_transient_version)
+        upgraded = self.longhorn.upgrade(upgrade_to_transient_version, timeout)
         if not upgraded:
             logging(f"Upgrading Longhorn failed")
-            time.sleep(self.retry_count)
-            assert False, "Upgrading Longhorn failed"
+            time.sleep(self.retry_interval)
+            return False
```

Comment:
It is modified to simply return the result to the upper layer (in this case, `longhorn.resource`), so the responsibility of evaluating the result now belongs to `longhorn.resource`.

For example,
```
logging(f"Upgrading Longhorn failed")
time.sleep(self.retry_count)
assert False, "Upgrading Longhorn failed"
```
should be moved to `longhorn.resource`

Reply:
Updated and [result](https://ci.longhorn.io/job/private/job/longhorn-e2e-test/3428/)